libs/editor/egui_editor/src/editor.rs

```diff
@@ -28,6 +29,14 @@ pub struct EditorResponse {
     pub has_selection: bool,
     pub edit_menu_x: f32,
     pub edit_menu_y: f32,
+
+    pub cursor_in_heading: bool,
+    pub cursor_in_bullet_list: bool,
+    pub cursor_in_number_list: bool,
+    pub cursor_in_todo_list: bool,
+    pub cursor_in_bold: bool,
+    pub cursor_in_italic: bool,
+    pub cursor_in_inline_code: bool,
```

Comment:
I'm not sure what should be in the editor response and what should be a function invoked via ffi but this is easier and looks clean enough

Reply:
Yes I tried to keep it clean. Since this struct is returned through ffi to Swift, I kept it simple and just a series of booleans. This is also 1 to 1 with how I represent what markdown element button is pressed on the swift side as well.